ZipBuilder/Sources/ZipBuilder/FrameworkBuilder.swift

```diff
@@ -406,32 +418,68 @@ struct FrameworkBuilder {
+      // TODO: Proper handling of podspec-specified module.modulemap files with customized umbrella
+      // headers. This is good enough for Firebase since it doesn't need these modules.
+      umbrellaHeader = "\(framework)-umbrella.h"
+    } else {
+      var umbrellaHeaderURL: URL
+      do {
+        let files = try fileManager.contentsOfDirectory(at: headersDir,
+                                                        includingPropertiesForKeys: nil).compactMap { $0.absoluteString }
+        let umbrellas = files.filter { $0.contains("umbrella.h") }
+        if umbrellas.count != 1 {
+          fatalError("Did not find exactly one umbrella header in \(headersDir).")
+        }
+        umbrellaHeaderURL = URL(string: umbrellas[0])!
+      } catch {
+        fatalError("Error while enumerating files \(headersDir): \(error.localizedDescription)")
+      }
+      // Verify Firebase frameworks include an explicit umbrella header for Firebase.h.
+      if framework.hasPrefix("Firebase"),
+        framework != "FirebaseCoreDiagnostics",
+        framework != "FirebaseUI",
+        !framework.hasSuffix("Swift") {
+        // Delete CocoaPods generated umbrella and use pre-generated one.
+        do {
+          try fileManager.removeItem(at: umbrellaHeaderURL)
+        } catch let error as NSError {
+          print("Failed to delete: \(umbrellaHeaderURL). Error: \(error.domain)")
+        }
+        umbrellaHeader = "\(framework).h"
+        let frameworkHeader = headersDir.appendingPathComponent(umbrellaHeader)
+        guard fileManager.fileExists(atPath: frameworkHeader.path) else {
+          fatalError("Missing explicit umbrella header for \(framework).")
+        }
+      } else {
+        umbrellaHeader = umbrellaHeaderURL.lastPathComponent
       }
     }
-    // Copy the Headers over. Pass in the prefix to remove in order to generate the relative paths
-    // for some frameworks that have nested folders in their public headers.
+    // Copy the Headers over.
     let headersDestination = frameworkDir.appendingPathComponent("Headers")
     do {
-      try recursivelyCopyHeaders(from: headersDir, to: headersDestination)
+      try fileManager.copyItem(at: headersDir, to: headersDestination)
     } catch {
       fatalError("Could not copy headers from \(headersDir) to Headers directory in " +
         "\(headersDestination): \(error)")
     }
 
+    // TODO: copy PrivateHeaders directory as well if it exists. SDWebImage is an example pod.
+
     // Move all the Resources into .bundle directories in the destination Resources dir. The
     // Resources live are contained within the folder structure:
     // `projectDir/arch/Release-platform/FrameworkName`
-    let arch = Architecture.arm64
-    let contentsDir = projectDir.appendingPathComponents([arch.rawValue,
-                                                          "Release-\(arch.platform.rawValue)",
+
+    let contentsDir = projectDir.appendingPathComponents([anyArch.rawValue,
```

Comment:
Nit: I would suggest not to use `rawValue` directly but add a method with self-explanatory name to the enum to return it. It will improve readability and simplify modification in the future. 

It's probably outside of the scope of the PR. I just would like to bring this topic up to consider in the future.